### ISSUE TITLE
Release version 0.1

### ISSUE THREAD

**Issue by leotrs** [ManimCommunity/manim#148]
There seems to be enough activity on this repo to try and move forward with a release version of the community version of manim. For this, let's use this issue to keep track of what needs to be done and when. Logistically, this should be the last issue to be closed in the v1.0.0 milestone, and once this one is closed, we should be able to push to pypi.

# Things that need to be discussed

1. I think there is a sense among @ManimCommunity/core devs that v1.0.0 should be fully backwards compatible. This has not been fully discussed, however.
2. For full backwards compatibility, we need to [determmine a timeline for deprecation](#145).
3. It has been decided that this repo will have both `manim` and `manimcm` as entrypoints. Some of us have started calling it manim CE (for Community Edition). Personally, I like "manim CE" as a name (and the accompanying `manimce` entrypoint) more than "manimcm". Though I don't ultimately care either way...
4. I think it would be nice to have a logo at the time of release #106 

# Things that need to be implemented
~Get rid of `CONFIG` dicts in each class and migrate to dataclasses #7~
- [x] ~~Add extensive tests #133~~
- [x] ~~Add documentation #103 #38 #104 with sphinix on ReadTheDocs #38, including a quickstart #120~~
- [x] ~~Migrate configuration to `configparser` and cascading `.cfg` files #98.~~
- [x] ~~Clean up `constants.py` #98.~~
- [x] Make a logo #106 
- [x] There are a bunch of things that depend on #7, ~~#98~~, and ~~#133~~, so we have to determine whether each one of those are required for a v1.0.0.
- [x] ~~Add a Scene Cacher #166~~
- [x] Additional documentation examples ~~#458~~ 
- [x] Various small improvements #489, ~~#490~~, ~~#492~~, ~~#496~~, ~~#498~~ 
- [x] Automate publishing to pypi ~~#500~~
- [ ] #572

Please feel free to edit this issue to add/remove/check off things as you see fit.

# After the release
- [ ] make PangoText the default backend of Text (#487)
- [ ] invite developers of other manim-related projects to become part of the organization (or at least ask them to support manim CE). For example: https://github.com/krassowski/jupyter-manim

**Comment by yoshiask** [comment]
To be clear, new features like the improved SVG parser and more GraphScene configuration are not slated for the 1.0.0 release?

**Comment by leotrs** [comment]
Yeah this is exactly what we should discuss here. My personal opinion is that v1.0.0 should be fully backwards compatible, as well as documented and tested, and some of the major problems with the code should be addressed. If that's the case, then a v1.0.0 could be ready in perhaps a couple of months. We could then leave new features for v1.1 release.

Now, if there are new features ready for a v1.0.0 I guess there's no hurt in adding them as long as they are fully backwards compatible. But the intent of this versioning is that most focus should be put into finishing v1.0.0 (i.e. in testing and documenting) before spending our efforts in new features.

**Comment by leotrs** [comment]
Argh, good question. It does support all the same flags (and fixes some of them too), but many variables from `constants.py` are removed. So no, it's not compatible.

**Comment by PgBiel** [comment]
by thursday I should open the draft PR on attrs, typings and docs

**Comment by huguesdevimeux** [comment]
@leotrs Could you update this with you work on #98 ?

**Comment by leotrs** [comment]
Just updated the check list at the top. Please refer to it for what to do next.

**Comment by leotrs** [comment]
As far as I can tell from yesterday's Discord call, we will try to have the following ready for a 1.0 release:

1. documentation up on RTD (#381 )
2. pypi package (#202 #202 )
3. logo

**Comment by naveen521kk** [comment]
@leotrs We are using poetry for releasing and stuff. So do add it in list also.

**Comment by leotrs** [comment]
Good call

**Comment by huguesdevimeux** [comment]
(There is a milestone for the release, let's use it ;) )

**Comment by naveen521kk** [comment]
Another thing is we need to change `manimlib` to `manim` in `pyproject.toml` 
#165

**Comment by leotrs** [comment]
Folks, it seems we have completed every item on the list except for the logo, which should be decided upon by Tuesday Oct 20th noon GMT. I suggest we add the logo and also fix #407 and then release this bad boi. Objections? Comments? @ManimCommunity/core 

If there are no objections, I move we put a freeze on PRs right now.
